mist-api/src/lib/util.js

```diff
@@ -0,0 +1,58 @@
+'use strict'
+
+// Vendor
+const _ = require('lodash')
+
+// Local
+const util = require('core-lib/util')
+
+exports.splitIntoQueryTerms = (value) => {
+    return util.splitIntoTerms(value).map((term) => `%${term}%`)
+}
+
+const modelNamesForOptimization = ["Gene"]
+
+exports.processSearch = (queryValue, target, modelName, textFields, exactMatchFields = null) => {
+    if (!queryValue)
+        return
+    if (!target)
+        throw new Error('processSearch must have a defined target')
+
+    const terms = exports.splitIntoQueryTerms(queryValue)
+    if (terms.length > 0) {
+        if (textFields)
+            textFields
+            .forEach((fieldName) => {
+                _.set(target, `criteria.where.$or.${fieldName}.$ilike.$all`, terms)
+            })
+
+        if (modelName && modelNamesForOptimization.includes(modelName)) {
+            // don't bother searching against other fields if search term is most probably a [gene] product.
+            if (exactMatchFields && terms.length == 1)
```

Comment:
Please take one more look at the routine.

"I don't think that it is safe to assume that if there is only one search term the user wants to solely search just the gene product field" - the situation is the other way around: 

at the top you see that the text search will be initiated regardless of the length of the term (if it's > 0): ` if (textFields) ...`
Inside:
if the model name is "Gene" (in modelNamesForOptimization array) and the term length == 1 then search against exact match fields (with id's of different flavors, they are one-word in length);

otherwise (or if model name is not provided) then search against exact match fields regardless of the terms length.  

So, its valid and safe. And currently only for genes this should be done. Moreover considering that the number of genes in the database is substantial this can be quite beneficial.